lib/Runtime/Library/JavascriptBigInt.cpp

```diff
@@ -0,0 +1,205 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#include "RuntimeLibraryPch.h"
+
+namespace Js
+{
+    Js::Var JavascriptBigInt::ToVar(const char16 * content, charcount_t cchUseLength, bool isNegative, ScriptContext * scriptContext, JavascriptBigInt *result)
+    {
+        AssertMsg(result != NULL, "Cannot use InPlaceNew without a value result location");
+        result->InitFromCharDigits(content, cchUseLength, isNegative);
+        return reinterpret_cast<Var>(result);
+    }
+
+    Var JavascriptBigInt::NewInstance(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        PROBE_STACK(function->GetScriptContext(), Js::Constants::MinStackDefault);
+
+        ARGUMENTS(args, callInfo);
+        ScriptContext* scriptContext = function->GetScriptContext();
+
+        AssertMsg(args.HasArg(), "Should always have implicit 'this'");
+
+        // SkipDefaultNewObject function flag should have prevented the default object from
+        // being created, except when call true a host dispatch.
+        Var newTarget = args.GetNewTarget();
+        bool isCtorSuperCall = JavascriptOperators::GetAndAssertIsConstructorSuperCall(args);
+
+        JavascriptBigInt* result = nullptr;
+
+        if (args.Info.Count > 1)
+        {
+            result = JavascriptConversion::ToBigInt(args[1], scriptContext);
+        }
+        else
+        {
+            // TODO:
+            // v8 throw: cannot convert from undefined to bigint
+            // we can consider creating a Zero BigInt
```

Comment:
spec says we have to throw a TypeError

Reply:
If you'd rather not add the throw now, then please crash the program (`AssertOrFailFast(false)`) to avoid any surprises that might result from returning nullptr

---
In reply to: [218222969](https://github.com/Microsoft/ChakraCore/pull/5706#discussion_r218222969) [](ancestors = 218222969)

Reply:
same goes for next block below

---
In reply to: [218223798](https://github.com/Microsoft/ChakraCore/pull/5706#discussion_r218223798) [](ancestors = 218223798,218222969)

Reply:
use `AssertOrFailFast(false)` for now